packages/react-client/src/ReactFlightClient.js

```diff
@@ -1003,6 +1003,24 @@ function getOutlinedModel<T>(
       let value = chunk.value;
       for (let i = 1; i < path.length; i++) {
         value = value[path[i]];
+        if (value.$$typeof === REACT_LAZY_TYPE) {
+          const referencedChunk: SomeChunk<any> = value._payload;
+          if (referencedChunk.status === INITIALIZED) {
+            value = referencedChunk.value;
+          } else if (
+            referencedChunk.status === BLOCKED ||
+            referencedChunk.status === PENDING
+          ) {
+            return waitForReference(
+              referencedChunk,
+              parentObject,
+              key,
+              response,
+              map,
+              path.slice(i),
+            );
+          }
```

Comment:
We should probably handle `RESOLVED_MODEL`, `RESOLVED_MODULE`, and `ERRORED` as well here. 🤔

Reply:
Yea it’s basically anything elser